### ISSUE TITLE
CodeLayout::ProhibitParensWithBuiltins reports eof()

### ISSUE THREAD

**Issue by vaitkus** [Perl-Critic/Perl-Critic#941]
### Issue description
CodeLayout::ProhibitParensWithBuiltins module, v1.138, reports the call to `eof` builtin with empty parentheses (`eof()`) although this construct has a special meaning. Citing [1]:
```
An eof without an argument uses the last file read.
Using eof() with empty parentheses is different.
It refers to the pseudo file formed from the files listed
on the command line and accessed via the <> operator.
```
### Steps to reproduce the issue
Run perlcritic on the following snippet:
```
if (eof()) {
    exit;
}
```
### Expected behaviour
The eof() construct is recognized as special and silently ignored.

[1] https://perldoc.perl.org/functions/eof

**Comment by petdance** [comment]
So you're saying that the policy should not complain about `eof` without parens?

**Comment by vaitkus** [comment]
> So you're saying that the policy should not complain about `eof` without parens?

Not exactly. `eof` without parentheses ("`eof`") is already ignored by the policy (as expected). I suggest that the policy should also not complain about `eof` with empty parentheses ("`eof()`"), that is, `eof` with no explicit parameters.

**Comment by petdance** [comment]
Right, sorry, I got that backwards.  It ignores `eof` because it doesn't have parens, but it shouldn't complain about `eof()` because there is a legit reason to have parens on it.
